src/integrationTest/java/org/opensearch/security/privileges/dlsfls/FlsDocumentFilterTest.java

```diff
@@ -0,0 +1,373 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+package org.opensearch.security.privileges.dlsfls;
+
+import com.google.common.collect.ImmutableSet;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class FlsDocumentFilterTest {
+
+    @Test
+    public void identity() throws Exception {
+        String sourceDocument = """
+            {
+                "big_integer": 12345678901234567890123456789012345678901234567890,
+                "string": "x",
+                "big_float": 12345678901234567890123456789012345678901234567890.123456789,
+                "object": {
+                   "attribute": "x",
+                   "nested_object": {
+                      "x": "y"
+                   },
+                   "nested_array": [1,2,3]
+                },
+                "array": [
+                   1,
+                   "x",
+                   {
+                      "foo": "bar"
+                   },
+                   [1,2,3,4]
+                ]
+            }
+            """;
+
+        byte[] result = FlsDocumentFilter.filter(
+            sourceDocument.getBytes(UTF_8),
+            FieldPrivileges.FlsRule.ALLOW_ALL,
+            FieldMasking.FieldMaskingRule.ALLOW_ALL,
+            ImmutableSet.of()
+        );
+
+        assertJsonStructurallyEquivalent(sourceDocument, result);
+    }
+
+    @Test
+    public void filterSimpleAttribute_exclusion() throws Exception {
+        String sourceDocument = """
```

Comment:
nit: I love using text blocks as well, but it will make the backport fail since 2.x supports lower versions of java. This PR will likely need a manual backport, but figured I would leave a comment about this.

Reply:
Good point, I will replace this by simple strings.